### ISSUE TITLE
Chapter 3 Example code not correct

### ISSUE THREAD

**Issue by andradei** [stevedonovan/gentle-intro#31]
The file _3-filesystems.md_ has a few bad code examples (won't compile) right in the beginning:

```rust
fn read_all_lines(filename: &str) -> io::Result<()> {
    let file = File::open(&filename)?;

    let mut lines = Lines::new(file);
    while let Some(line) = lines.next() {
        let line = line?;
        println!("{}", line)?;
    }

    Ok(())
}
```

There, `println!("{}", line)?;` won't compile because `println!` returns `()` and not `Result`.

The same happens on the very next code example:

```rust
while let Some(Ok(line)) = lines.next() {
        println!"{}", line)?;
    }
```

Which even has a parenthesis missing.

**Comment by stevedonovan** [comment]
I'm embarrassed ;)  Ok, from now on, _nothing_ goes into book that hasn't passed through `rustc`. `mdbook` has a test facility which I'm going to investigate.

**Comment by stevedonovan** [comment]
Fixed by last push - sorry didn't mention #31
